Detectors/TRD/reconstruction/src/CruRawReader.cxx

```diff
@@ -357,6 +357,7 @@ int CruRawReader::parseDigitHCHeader()
           //LOG(alarm) << "Digit HC Header 1 reserved : " << std::hex << mDigitHCHeader1.res << " raw: 0x" << mDigitHCHeader1.word;
           increment2dHist(TRDParsingDigitHeaderWrong1, mFEEID.supermodule * 2 + mHalfChamberSide[0], mStack[0], mLayer[0]);
         }
+        mTimeBins = mDigitHCHeader1.numtimebins;
```

Comment:
I might be paranoid, but I'd suggest a check here if the number of timebins from the HCH1 is the same as in the preset. We could use a preset of -1 to indicate that we trust the HC1.

Reply:
mmm good point. So you dont trust the contents of the headers ;-)

Reply:
24 and 27 wont matter so long as its <30. It does not even have to be a multiple of 3, although I am not sure what the trap will do with a number not divisible by 3. Of course if its >30 then those adc values are going to get dumped after the first one with >30 up to the next digit end marker. Paranoid is sometimes bad, for this type of code, running on line paranoid is the wiser stance.